src/devtools/client/inspector/ResponsiveTabs.tsx

```diff
@@ -0,0 +1,175 @@
+import cx from "classnames";
+import React, {
+  CSSProperties,
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+
+type ResponsiveTabsProps = {
+  className?: string;
+  dropdownButton?: ReactNode;
+  dropdownClassName?: string;
+  dropdownStyle?: CSSProperties;
+  children: ReactElement | ReactElement[];
+  activeIdx: number;
+};
+
+export const ResponsiveTabs = ({
+  className = "",
+  dropdownButton,
+  dropdownClassName,
+  dropdownStyle,
+  children,
+  activeIdx,
+}: ResponsiveTabsProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const tabsRef = useRef<HTMLSpanElement[]>([]);
+  const dropdownRef = useRef<HTMLSpanElement>(null);
+
+  const tabs = Array.isArray(children) ? children : [children];
+
+  const [visibleItemsCount, setVisibleItemsCount] = useState(tabs.length);
+  const [indicesOrder, setIndicesOrder] = useState<number[]>(
+    Array.from({ length: tabs.length }).map((_, idx) => idx)
+  );
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  const recalcTabsVisibilityAndOrder = useCallback(() => {
+    const containerWidth = containerRef.current?.clientWidth ?? 0;
+
+    // we will always close dropdown when resizing
+    setDropdownOpen(false);
+
+    // check if maybe we don't need a dropdown at all
+    let totalTabsWidth = 0;
+    for (const tab of tabsRef.current) {
+      totalTabsWidth += tab.clientWidth;
+    }
+
+    if (totalTabsWidth <= containerWidth) {
+      setVisibleItemsCount(tabsRef.current.length);
+      setIndicesOrder(Array.from({ length: tabs.length }).map((_, idx) => idx));
+      return;
+    }
+
+    // we will need a dropdown for sure,
+    // so let's calc which items are visible
+    const dropdownButtonWidth = dropdownRef.current?.clientWidth ?? 0;
+    const activeTabWidth = tabsRef.current[activeIdx].clientWidth ?? 0;
+
+    let visibleCount = 1;
+    let tabIndices: number[] = [activeIdx];
+
+    let runningWidth = dropdownButtonWidth + activeTabWidth;
+    for (let idx = 0; idx < tabs.length; idx++) {
+      if (idx === activeIdx) {
+        continue;
+      }
+
+      runningWidth += tabsRef.current[idx].clientWidth;
+
+      if (runningWidth <= containerWidth) {
+        if (idx < activeIdx) {
+          tabIndices.pop();
+          tabIndices = [...tabIndices, idx, activeIdx];
```

Comment:
I'd prefer `tabIndices.splice(tabIndices.length - 1, 0, idx)`, so that it's more obvious that you're just inserting `idx` at a specific position.